commands/cloudapi-v6/nic.go

```diff
@@ -1033,3 +1044,151 @@ func DetachAllNics(c *core.CommandConfig) error {
 	fmt.Fprintf(c.Command.Command.OutOrStdout(), jsontabwriter.GenerateLogOutput("Nics successfully detached from Load Balancer"))
 	return nil
 }
+
+func validateIPv6CidrBlockForNIC(cidr string, lan ionoscloud.Lan) error {
+	ip, ipNet, err := net.ParseCIDR(cidr)
+	if err != nil {
+		return err
+	}
+
+	if ip.To4() != nil {
+		return fmt.Errorf("this is not an IPv6 Cidr Block")
+	}
+
+	if ones, _ := ipNet.Mask.Size(); ones != 80 {
+		return fmt.Errorf("invalid network mask for the provided IPv6 Cidr Block")
+	}
+
+	lanProperties, ok := lan.GetPropertiesOk()
+	if !ok || lanProperties == nil {
+		return fmt.Errorf("could not retrieve LAN properties")
+	}
+
+	lanIPv6CidrBlock, ok := lanProperties.GetIpv6CidrBlockOk()
+	if !ok {
+		return fmt.Errorf("could not retrieve LAN IPv6 Cidr Block")
+	}
+	if lanIPv6CidrBlock == nil {
+		return fmt.Errorf("IPv6 is not enabled on this LAN")
+	}
+
+	_, lanIPNet, err := net.ParseCIDR(*lanIPv6CidrBlock)
+	if err != nil {
+		return err
+	}
+
+	if !lanIPNet.Contains(ip) {
+		return fmt.Errorf("the provided LAN IPv6 Cidr Block is not within the LAN IPv6 Cidr Block")
```

Comment:
I don't understand this error, do you mean `the provided IP is not within the LAN IPv6 Cidr Block range`?